packages/client/lib/miner/pendingBlock.ts

```diff
@@ -58,9 +89,29 @@ export class PendingBlock {
     withdrawals?: WithdrawalData[]
   ) {
     const number = parentBlock.header.number + BigInt(1)
-    const { timestamp } = headerData
+    const { timestamp, mixHash } = headerData
     const { gasLimit } = parentBlock.header
 
+    // payload is uniquely defined by timestamp, gasLimit and the header
+    const timestampBuf = bigIntToUnpaddedBuffer(toType(timestamp ?? 0, TypeOutput.BigInt))
+    const gasLimitBuf = bigIntToUnpaddedBuffer(gasLimit)
+    const mixHashBuf = toType(mixHash!, TypeOutput.Buffer) ?? zeros(32)
```

Comment:
Doesn't this crash if `mixHash` is `undefined`?

Reply:
mixhash (prevRandao) is a mandatory field in fcU